gufe/settings/models.py

```diff
@@ -12,16 +12,53 @@
 from openff.models.types import FloatQuantity
 from openff.units import unit
 from pydantic import Extra, Field, PositiveFloat
+from pydantic import PrivateAttr
 
 
 class SettingsBaseModel(DefaultModel):
     """Settings and modifications we want for all settings classes."""
+    _is_frozen: bool = PrivateAttr(default_factory=lambda: False)
 
     class Config:
         extra = Extra.forbid
         arbitrary_types_allowed = False
         smart_union = True
 
+    def frozen_copy(self):
+        copied = self.copy(deep=True)
+
+        def freeze_model(model):
+            submodels = (
+                mod for field in model.__fields__
+                if isinstance(mod := getattr(model, field), SettingsBaseModel)
+            )
+            for mod in submodels:
+                freeze_model(mod)
+
+            if not model._is_frozen:
+                model._is_frozen = True
+
+        freeze_model(copied)
+        return copied
+
+    def __setattr__(self, name, value):
+        if self._is_frozen:
+            raise TypeError(f"Cannot set '{name}': Settings are immutable "
```

Comment:
For the sake of your regular non-programmer user, I'd recommend toning down the word immutable to "can't be changed" or something?